src/lib/DateInput.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+  import { v4 as uuidv4 } from "uuid";
+  import { DefaultButton, ErrorMessage, FormElement } from "./index.js";
+
+  export let label: string;
+  export let errorMessage = "";
+  export let setToTodayButton = false;
+  export let value: string;
+
+  let id = uuidv4();
+  let dayId = `${id}-day`;
+  let monthId = `${id}-month`;
+  let yearId = `${id}-year`;
+
+  let dayString = "";
+  let monthString = "";
+  let yearString = "";
+  $: {
+    errorMessage = "";
+    const dayInCorrectRange =
+      parseInt(dayString) > 0 || parseInt(dayString) < 32;
+    const monthInCorrectRange =
+      parseInt(monthString) > 0 && parseInt(monthString) < 13;
+    const yearCorrectLength = yearString.length === 4;
+    if (!dayInCorrectRange && dayString !== "") {
+      errorMessage = "Day must be a number between 1 and 31. ";
+    }
+    if (!monthInCorrectRange && monthString !== "") {
+      errorMessage += "Month must be a number between 1 and 12. ";
+    }
+    if (!yearCorrectLength && yearString !== "") {
+      errorMessage += "Year must be a number of length 4.";
+    }
+
+    if (dayInCorrectRange && monthInCorrectRange && yearCorrectLength) {
+      monthString = ("0" + monthString).slice(-2);

```

Comment:
You'd also want to pad the day...

What if you start this block by parsing Y, M, and D as numbers, then check those're in the correct range, and then stringify them to make `value`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart could help with padding, unless JS has the `%02d` type sprintf somewhere